File: docs/competitors/lab1/php/FizzBuzz.php
<?php
declare(strict_types=1);

define('INPUT_MESSAGE', "Введите целое число (для завершения введите 'stop')");
define('ERROR_MESSAGE', "Пожалуйста, введите целое число или 'stop' для завершения");
define('INPUT_SYMBOL', ": ");
define('STOP_WORD', "stop");
define('DIVISIBLE_BY_3', "Fizz");
define('DIVISIBLE_BY_5', "Buzz");
define('NEW_LINE', "\n");

function main() 
{
    echo INPUT_MESSAGE . NEW_LINE;
    while (true)
    {
        echo INPUT_SYMBOL;
        $input = trim(fgets(STDIN));

        if (strtolower($input) === STOP_WORD)
        {
            break;
        }

        if (validateInput($input)) 
        {
            echo ERROR_MESSAGE . NEW_LINE;
            continue;
        }

        $inputNumber = intval($input);
        echo defineResultMessage($inputNumber);
    }
}

function validateInput(string $input): bool
{
    return empty($input) 
        || !is_numeric($input) 
        || strpos($input, '.') !== false;
}

function defineResultMessage(int $number): string
{
    $canBeDivided3 = canBeDivided($number, 3);
    $canBeDivided5 = canBeDivided($number, 5);
    $resultMessage = "";

    if ($canBeDivided3) 
    {
        $resultMessage .= DIVISIBLE_BY_3;
    } 
    
    if ($canBeDivided5) 
    {
        $resultMessage .= DIVISIBLE_BY_5;
    } 

    if (!$canBeDivided3 && !$canBeDivided5) 
    {
        $resultMessage .= $number;
    }

    return $resultMessage .= NEW_LINE;
}

function canBeDivided(int $dividend, int $divider): bool
{
    return $dividend % $divider === 0;
}

main()
?>

File: docs/competitors/lab1/go/FizzBuzz/FizzBuzz.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

const (
	firstMessage = "Введите целое число (для завершения введите 'stop')"
	errorInputMessage = "Пожалуйста, введите целое число или 'stop' для завершения"
	inputSymbol  = ": "
	stopWord     = "stop"

	newLineSymbol = "\n"
	divisibleBy3  = "Fizz"
	divisibleBy5  = "Buzz"
)

func main() {
	fmt.Println(firstMessage)
	scanner := bufio.NewScanner(os.Stdin)

	for {
		fmt.Print(inputSymbol)
		scanner.Scan()
		input := strings.TrimSpace(scanner.Text())

		if strings.ToLower(input) == stopWord {
			break
		}

		if validateInput(input) {
			fmt.Println(errorInputMessage)
			continue
		}

		num, _ := strconv.Atoi(input)
		fmt.Print(defineResultMessage(num))
	}
}

func validateInput(input string) bool {
	_, err := strconv.Atoi(input)
	return err != nil 
}

func defineResultMessage(number int) string {
	dividedBy3 := canBeDivided(number, 3)
	dividedBy5 := canBeDivided(number, 5)
	resultMessage := ""

	if dividedBy3 {
		resultMessage += divisibleBy3
	}

	if dividedBy5 {
		resultMessage += divisibleBy5
	}

	if (!dividedBy3 && !dividedBy5) {
		resultMessage += strconv.Itoa(number)
	}

	resultMessage += newLineSymbol
	return resultMessage
}

func canBeDivided(dividend, divider int) bool {
	return dividend%divider == 0
}